skypy/galaxy/spectrum.py

```diff
@@ -122,99 +122,135 @@ def dirichlet_coefficients(redshift, alpha0, alpha1, z1=1., weight=None):
     ----------
     .. [1] M. R. Blanton et al., 2003, AJ, 125, 2348
 
-    Examples
-    --------
-    Get B-band magnitudes for the kcorrect spec templaces using auto-loading
-    of known spectral data:
-    >>> from skypy.galaxy.spectrum import mag_ab
-    >>> mag_B = mag_ab('kcorrect_spec', 'Johnson_B')  # doctest: +SKIP
-
     '''
 
-    # get the spectra and bandpasses
-    spec_lam = spectrum.wavelength.to_value(units.AA, equivalencies=units.spectral())
-    spec_flux = spectrum.flux.to_value('erg s-1 cm-2 AA-1',
-                                       equivalencies=units.spectral_density(spec_lam))
-    band_lam = bandpass.wavelength.to_value(units.AA, equivalencies=units.spectral())
-    band_tx = bandpass.flux.to_value(units.dimensionless_unscaled)
-
-    # redshift zero if not given
-    if redshift is None:
-        redshift = 0.
-
-    # Array shapes
-    nz_loop = np.atleast_1d(redshift).shape
-    ns_loop = np.atleast_2d(spec_flux).shape[:-1]
-    nb_loop = np.atleast_2d(band_tx).shape[:-1]
-    nz_return = np.shape(redshift)
-    ns_return = spec_flux.shape[:-1]
-    nb_return = band_tx.shape[:-1]
-    loop_shape = (*nz_loop, *nb_loop, *ns_loop)
-    return_shape = (*nz_return, *nb_return, *ns_return)
-
-    # allocate magnitude array
-    mag_ab = np.empty(loop_shape, dtype=float)
-
-    # compute magnitude contribution from band normalisation [denominator of (2)]
-    m_band = -2.5*np.log10(np.trapz(band_tx/band_lam, band_lam))
-
-    # magnitude offset from band and AB definition [constant in (2)]
-    m_offs = -2.4079482426801846 - m_band
-
-    # compute flux integrand at emitted wavelengths
-    spec_intg = spec_lam*spec_flux
-
-    # go through redshifts ...
-    for i, z in enumerate(np.atleast_1d(redshift)):
+    # load the filters
+    if np.ndim(filters) == 0:
+        filters = (filters,)
+    filters = speclite.filters.load_filters(*filters)
+    if np.shape(filters) == (1,):
+        filters = filters[0]
 
-        # observed wavelength of spectra
-        obs_lam = (1 + z)*spec_lam
+    # number of dimensions for each input
+    nd_s = len(np.shape(spectrum)[:-1])  # last axis is spectral axis
+    nd_f = len(np.shape(filters))
+    nd_z = len(np.shape(redshift))
 
-        for j, b in enumerate(np.atleast_2d(band_tx)):
+    # check if interpolation is necessary
+    if interpolate and np.size(redshift) <= interpolate:
+        interpolate = False
 
-            # interpolate band to get transmission at observed wavelengths
-            obs_tx = np.interp(obs_lam, band_lam, b, left=0, right=0)
-
-            # compute magnitude contribution from flux [numerator of (2)]
-            mag_ab[i, j, :] = -2.5*np.log10(np.trapz(spec_intg*obs_tx, obs_lam))
-
-    # combine AB magnitude [all of (2)]
-    mag_ab += np.atleast_1d(m_offs)[:, np.newaxis]
+    # if interpolating, split the redshift range into `interpolate` bits
+    if interpolate:
+        redshift_ = np.quantile(redshift, np.linspace(0, 1, interpolate))
```

Comment:
```suggestion
        redshift_ = np.linspace(np.min(redshift), np.max(redshift), interpolate)
```
Given that galaxies are not uniform in redshift, I'm not sure which of these would work better in general?

Reply:
That is the reason I changed it to produce equally populated redshift slices.

Reply:
We discussed this offline and concluded that both are reasonable approaches and the difference between them should be small. Accuracy can always be improved with more interpolation points.